### ISSUE TITLE
RFC: Landcover (polygon hazard on polygon exposure)

### ISSUE THREAD

**Issue by akbargumbira** [inasafe/inasafe#1216]
# Problem

There is no land cover impact function in InaSAFE. Understanding the impacts of a hazard on landcover is important because we need to know the effects of a disaster on peoples livelihood assets (especially the crops) inside the land cover and food security.

![landcover - new page](https://cloud.githubusercontent.com/assets/178003/4368452/3b6a2178-42ed-11e4-899d-2db09d2d038f.png)

# Duration

TBD
# Status

This RFC is in **draft**.

# Proposers

@mbernasocchi 
@timlinux

# Detail

Land cover has some specific metrics we are interested in:

1. **Type of Land Cover**. This is a measure of what type of landcover affected by hazards
2. **Area of Land Cover**. This is a measure how many extents/areas affected by hazards

Specific for crops, we are also interested in : 

4. **Productivity**: This is a measure of crop yield e.g. 1000 kg rice / ha / annum.
5. **Economic value**: This is a measure of crop value e.g. Rp 100000 / rice / ha / annum.
6. **Area**. This is a measure of land cover/use total area e.g. 1000 ha or 100 m2
7. **type of plants**. This is a information what kind of crops existing or productive e.g rice, corn, etc
8. **harvested area**. This is a measure of productive crops area e.g 100 Ha or 100 Km2

We can treat these metrics using generalised assumptions in the same way gender ratios are done but there are differences in approach. In the first instance gender break down is based on post-processing of the impact with aggregation layer, whereas in this use case we would be post processing the impact layer. We have a problem though that it does not make sense to apply global defaults (equivalent to e.g. national gender ratio) as prices of crops and goods may vary greatly across the country. Is this assertion correct?

## Where do we find appropriate land use data?

* BIG provides land use data for Indonesia - this data can be downloaded free trough : inageoportal
  
http://portal.ina-sdi.or.id/faces/oracle/webcenter/portalapp/pagehierarchy/Page49.jspx?_afrLoop=30041381169853277&_afrWindowMode=0&_afrWindowId=null#%2Foracle%2Fwebcenter%2Fportalapp%2Fpagehierarchy%2FPage35.jspx%40%3F_adf.ctrl-state%3Dnq2gr7910_4
   
The data scale is 1 : 25.000 
These data attribute consist of  name of type of land cover, Fcode, and area (in m2)
The land cover spatial data production is under Center of Integration Thematic and Mapping-BIG. 

* General Directorate of Forestry Planning - The spatial and e-book data can be downloaded on KML format at http://appgis.dephut.go.id/appgis/download.aspx for each province. The Latest update that published is 2011 land cover. 
The spatial data attribute only consist of  name of the landuse. 
![atribute of landuse](https://cloud.githubusercontent.com/assets/6345285/5521406/31e6f384-89d2-11e4-9ff6-3fd30be9f9fb.PNG)

The data sources come from Directorate of Inventory and Monitoring Forestry Resources at 
General Directorate of Forestry Planning. In 1 : 250.000 scale. The data consist of 
1. Forestry Thematic Based Map (PDTK) in 1 : 250.000 scale
2. Land Cover digital data from 2012 Landsat 7 ETM+ interpretation. There are 23 land cover classification as output from the interpretation.

* Center of Statistic Agency (BPS-Badan Pusat Statistik) have the tabular data excel format for crops productivity, harvested area, type of plants, total production. What published in the website http://bps.go.id/tnmn_pgn.php?kat=3&id_subyek=53&notab=0
This kind of data can be integrated into spatial data

* Kementrian Pertanian Republik Indonesia (Ministry of agriculture) also have center of landuse data management based on WebGIS. Have a look at http://gis.pertanian.go.id
the official website also provide basis data of agriculture which can be download (free) in tabular format : http://aplikasi.pertanian.go.id/bdsp/newlok.asp
![harvest area tabular data_ministry of agriculture](https://cloud.githubusercontent.com/assets/6345285/5610541/cb40a754-94eb-11e4-89fa-f4a03bae3757.PNG)
  

# Proposed solution

Firstly we should mention that landuse is just a specific use case of polygon on polygon analysis. Our suggestion is that we should implement polygon (hazard) on polygon (exposure) analysis in a generic fashion and then implement post-processors that deal with specific use cases such as landuse.

- [ ] We should use basic data from BIG. BIG is mandatory agency that produce land cover in 1 : 25.000 scale. These data available to be downloaded freely

- [ ] Develop Land Cover Downloader. we can directly pull the land cover layers data(Shp) from http://portal.ina-sdi.or.id into InaSAFE. The mechanism might be similar with OSM downloader but we need to develop open-layer bounding box to choice specific area that exposed by hazard 

- [ ] Specific for crop analysis : Allow to join the tabular data (productivity, harvested area, type of plants & total production) from statistic or relevant agencies (agriculture or forestry) with land cover data that downloaded from Landcover downloader (http://portal.ina-sdi.or.id)

- [ ] Develop Key Questions in IF : 
       1. What type of land cover might be flooded/inundated (flood/tsunami IF) ; be covered by  
           volcano ash (Volcano IF) ; exposed by earthquake
       2. How many extents/area might be flooded/inundated (flood/tsunami IF) ; be covered by  
           volcano ash (Volcano IF) ; exposed by earthquake
    
- [ ]  Add tokens to the summary table data that can be incorporated into impact function report
- [ ] Port the categorical impact function from ESSC to InaSAFE main branch
- [ ]  We should ensure the impact function always works in a projected CRS so that area calculations are meaningful.
- [ ]  The impact function from ESSC should be updated to calculate areas.
- [ ]  Implement a new postprocessor for computing land use impact
- [ ]  Adapt the postprocessor to be able to optionally work on the impact layer (perhaps easy if we do post processing in the impact function directly we can do this easily).

# RFC Provenance

This RFC was discussed at the InaSAFE workshop in Jakarta, Tuesday 23 September 2014. The following people were in Attendance:

* Tim Sutton
* Akbar Gumbira
* Christian Christelis
* Charlotte Morgan
* Marco Bernasocchi
* Intan Pujawati (BIG Representative)
* Wulansari K
* Harry Mahardhika 
* Fredy Chandra
* Ismail Sunni
* Aditya Dwijananto
* Dianne Bencito

# Record of votes

+1 Tim Sutton
+1 Marco Bernasocchi
+1 Akbar Gumbira
+1 Fredy Chandra

# Resolution

# CC

**Comment by fredychandra** [comment]
I have added some information about land use data management and agriculture basis data availability from official website of Agriculture Ministry of Indonesia. This ministry have the mandate and authority to collect the agriculture data and send it to  Statistic agency to be published. I assume there should have the spatial layers (type of crops) consist of relevance attributes data for land cover analysis but it might be need some official approach.

**Comment by Charlotte-Morgan** [comment]
hi @mbernasocchi - I hope you don't mind. I have assigned this to @fredychandra as I want him to wrap this RFC up by the end of January - if not sooner.

**Comment by Charlotte-Morgan** [comment]
hi @fredychandra & @ismailsunni 
please note other related tickets & consolidate if possible
#1178 
#883 
#417 

@maning & @lianarazafy - based on your previous comments on #417 do you have any ideas / data sources / user requirements to add to this RFC?

# cc
@timlinux @mbernasocchi

**Comment by Charlotte-Morgan** [comment]
hi @fredychandra please prepare data for rice and flood and share with @ismailsunni so that he can look at this.

**Comment by fredychandra** [comment]
hi @ismailsunni please find and have a look the raw shapefile of lancover 25k and (Bengawan Solo) flood polygon. 

https://drive.google.com/folderview?id=0B_gC8poAYEtefmtMLVRHNEo4WndmbWR2RzlXY2Q1V0dxWWFtdlg2bmM5c2VQUU93Y2VOdWM&usp=sharing

The 25K landcover data is downloaded from BIG website, Ina-geoportal : 

http://portal.ina-sdi.or.id/faces/oracle/webcenter/portalapp/pagehierarchy/Page49.jspx?_afrLoop=31875830326354277&_afrWindowMode=0&_afrWindowId=null#%2Foracle%2Fwebcenter%2Fportalapp%2Fpagehierarchy%2FPage35.jspx%40%3F_adf.ctrl-state%3D8qaw4qlz_4

**Comment by fredychandra** [comment]
I just updated the newest land cover data prototype. This data have been clipped based on District Level (Kabupaten Bojonegoro), also that folder consist of Fcode Referense document. 

@ismailsunni @Charlotte-Morgan Please have a look on this link  : 

https://drive.google.com/folderview?id=0B_gC8poAYEtefjZTaHI5S3EtVUtENmhlQWFob0lkZDdpY2s2MC1YWnNEZzRCREdlUno0UWs&usp=sharing

**Comment by ismailsunni** [comment]
From discussion between @fredychandra, mas Iwan, and me. The things that need to be done first:

- [ ] Create IF polygon to polygon where the exposure has field for type of land (rice, corn, etc).
- [ ] Calculate the affected area by the hazard per type of land
- [ ] Calculate the percentage for each type of land

Additional notes:
It would be great if we can download data from http://portal.ina-sdi.or.id, but we can do it later after the IF is ready

**Comment by Charlotte-Morgan** [comment]
thanks @ismailsunni 
agree completely with above 3 steps. 
- [ ] calculate area of land (needs projected coordinates) will be the next step to be able to add values to each crop in post processing ... much later on

Agree - down load is a low priority, especially as data are not compiled into a single dataset on the portal.

cc @samnawi @fredychandra @iyan31

**Comment by timlinux** [comment]
@ismailsunni I am just wondering if this might be a good thing to delegate to @wonder-sk if he is interested and has time after doing his current tasks. I know your hours are nearly used up and it might be easy for Martin to implement - at least for the core spatial analysis parts.

**Comment by timlinux** [comment]
In experimental from @wonder_sk - needs more review - possibly we can merge this to develop for 3.3?

**Comment by samnawi** [comment]
@ismailsunni :+1 for you to make it happen all of the task in your list.

> - [ ] Create IF polygon to polygon where the exposure has field for type of land (rice, corn, etc).
- [ ] Calculate the affected area by the hazard per type of land
- [ ] Calculate the percentage for each type of land

also a task from @Charlotte-Morgan 

> - [ ] calculate area of land (needs projected coordinates) will be the next step to be able to add values to each crop in post processing

I've tried it last week, and this is the result.
![image](https://cloud.githubusercontent.com/assets/8871958/13452850/4287b94a-e07e-11e5-86c3-9ab259207110.png)

![image](https://cloud.githubusercontent.com/assets/8871958/13454944/fe183a5a-e08c-11e5-804a-01a6664bbab0.png)

I think this is a good step to make it (Landcover analysis) happen.

For the download data, as @Charlotte-Morgan mention if data are not compiled into a single dataset on the portal, and also in the attribute of the data, from many fields in there, only FCODE that filled.
![image](https://cloud.githubusercontent.com/assets/8871958/13454093/91b7d5be-e087-11e5-9674-e8efcf9c9a75.png)

REMARK column / field, should be filled by the landcover name. There is a standardization from BIG, but when I try to find it in BIG website, apparently their website can't be accessed (I'll update when the website is online).

# Proposed for next step

- Action list
- Notes
- If possibly, we can add post processing to add some parameters to make automatic impact calculation, especially for economic sector.

@fredychandra maybe you can add some comment from the result.

Thank you.

**Comment by fredychandra** [comment]
In think for first strike this analysis result is good. It provides basic information for how many total area of landcovers are affected by hazard. In the future, it would be nice if the result can only choose relevant existing landcover that impacted by hazard (depend of hazard type). For example for flood, the result should only show how many area of rice field (sawah), crops related landcover and settlement area (pemukiman) instead of show whole landcover type that not really necessary or relevant.

**Comment by fredychandra** [comment]
Please find the fcode description on the link below : 

https://drive.google.com/file/d/0B_gC8poAYEteNFVQQklrOVJCTXl3TklMWjduYkw2NGhCdEI4/view?usp=sharing

**Comment by samnawi** [comment]
As the agree from the last discussion, especially to support realtime volcanic ash #2491, I've already given @lucernae the landcover data at 1 : 250,000 scale.

For the remember, if we download the landcover data from http://tanahair.indonesia.go.id/home/, there are 3 scale level of the data. 
1 : 250,000 and 1 : 50,000 available for whole of country.
1 : 25,000 just available for Java, Bali, and Nusa Tenggara

At the 1 : 250,000 and 1 : 50,000 scale, already have the information in Remark column (not only FCODE)
**1 : 250,000**
![250k](https://cloud.githubusercontent.com/assets/8871958/14382440/f96487fa-fdb7-11e5-850e-19f053b39058.jpg)
**1 : 50,000**
![50](https://cloud.githubusercontent.com/assets/8871958/14382441/fb22acac-fdb7-11e5-9186-832f41b81982.jpg)

But for 1 : 25,000 scale, the information from the attribute just for FCODE
![25k](https://cloud.githubusercontent.com/assets/8871958/14382461/30d8485c-fdb8-11e5-80ff-cf2108253955.jpg)

Thank you

**Comment by wonder-sk** [comment]
For implementation of land cover IF - see #2709

**Comment by samnawi** [comment]
Hi @wonder-sk Is it correct if in the current version of develop, the landcover IF is just for the generic class in the hazard data?

Volcano hazard 
![image](https://cloud.githubusercontent.com/assets/8871958/14951194/56f015d4-1081-11e6-93a6-0d761dadebdf.png)

Flood hazard
![image](https://cloud.githubusercontent.com/assets/8871958/14951161/29a60a48-1081-11e6-890b-fab0129b3482.png)

Whether the landcover for specific hazard classes will available in 3.4 also?

cc :  @timlinux @ismailsunni @Charlotte-Morgan

**Comment by wonder-sk** [comment]
Currently the IF is indeed made to work with generic classes. I would be happy to update it to work also with specific hazards, however I am not sure what would be the correct approach (my understanding of IF metadata is a bit limited).

I have tried to change IF's metadata related to vector hazard classification - from ```[generic_vector_hazard_classes]``` to ```all_vector_hazard_classes``` but that seems to be not enough to make it work for flood hazards. Maybe @ismailsunni or @Gustry have some hints? Are there any routines that would allow me to handle a specific hazard layer as a generic one? Do the specific hazard types need some extra handling? Is it OK to allow all hazard types in an impact function?

**Comment by ismailsunni** [comment]
hi @wonder-sk 
Ok, I will take a look at that issue. It should be straight forward by design. But let me take a look first.

Meanwhile, I got a problem when I ran analysis by using test data in inasafe repo.
See image below for clearer explanation:
<img width="1303" alt="screen shot 2016-05-05 at 11 29 45" src="https://cloud.githubusercontent.com/assets/1421861/15036620/f77b1a4e-12bb-11e6-945e-3b6695ac227f.png">


For easy, I use single symbol for each layer. 

1. Red (50% transparency) for godzilla (hazard)
2. Green (50% transparency) for affected land cover (impact layer)
3. Blue (0% transparency) for land cover (exposure).

As we can see above, there 4 colours that I give label:

1. Blue : Exposure part that doesn't intersect with hazard layer.
2. Purple: Exposure intersect with Hazard
3. Brown:  Exposure intersect with hazard intersect with impact layer
4. Pale green (I don't know the colour name) : Exposure intersect with impact layer

It seems not right to me since there is an impact layer that is not overlap with hazard layer.

Any explanation about this?

cc @wonder-sk @Gustry

**Comment by Gustry** [comment]
I explained to @ismailsunni that the problem was about the wrong projection in the impact layer (different than 3857).

I'm working to make it work with more input data (tsunami raster)

**Comment by Gustry** [comment]
FYI, Tsunami on Landcover is in a PR : https://github.com/inasafe/inasafe/pull/2785

**Comment by timlinux** [comment]
This is implemented in 3.4 - I guess we want to look at other variants of 'on landcover' IF's in the future...
